h2o-core/src/main/java/water/parser/BufferedString.java

```diff
@@ -167,21 +171,22 @@ public void setOff(int off) {
     }
     return false;
   }
- 
-  public boolean sameString(String str) {
+
+  /**
+   * Tests whether this BufferedString is equal to a given ASCII string
+   * @param str string sequence made of ASCII characters (0..127)
+   * @return true if this BufferedString represents a given ASCII String, false if sequences differ or if the test
+   * string is not actually made of just ASCII characters
+   */
+  public boolean equalsAsciiString(String str) {
     if (str == null || str.length() != _len) return false;
-    for (int i = 0; i < _len; ++i)
-      if ((0xFF&_buf[_off + i]) != str.charAt(i)) return false;
-    return true;
-  }
-  
-  public boolean isOneOf(String[] samples) {
-    if (samples != null) {
-      for (String sample : samples) if (sameString(sample)) return true;
+    for (int i = 0; i < _len; ++i) {
+      char c = str.charAt(i);
+      if ((0x7F & c) != c || (_buf[_off + i]) != c) return false;
```

Comment:
Do we need the first condition here? Looks to me that if this is ascii, str better be ascii too or it will fail comparison on at least on of the bytes?